pkg/translate/field_mask.go

```diff
@@ -22,3 +24,84 @@ func parseFieldMask(message proto.Message, paths []string) (model.Mask, error) {
 
 	return parsedMask.Paths, nil
 }
+
+// updateModelFromFieldMask updates the dst with the values from src based on the fieldMask.
+//
+// Following scenarios are possible:
+//   - fieldMask is nil: All fields are updated.
+//   - fieldMask is not nil, but has no paths: No fields are updated.
+//   - fieldMask contains paths, that does not exist in the model: The paths are ignored.
+//   - fieldMask contains paths, that exist in model: Only those fields are updated.
+//
+// `protoName` tags are used to match fields from the fieldMask to fields in the model.
+func updateModelFromFieldMask[T any](mask model.Mask, dst, src *T) *T {
```

Comment:
It's better if src is first followed by dst.

```suggestion
func copyMask[T any](src, dst *T, mask model.Mask) *T {
```